src/librustdoc/core.rs

```diff
@@ -492,9 +492,9 @@ impl<'tcx> Visitor<'tcx> for EmitIgnoredResolutionErrors<'tcx> {
                 "could not resolve path `{}`",
                 path.segments
                     .iter()
-                    .map(|segment| segment.ident.as_str().to_string())
-                    .collect::<Vec<_>>()
-                    .join("::")
+                    .map(|segment| segment.ident.as_str())
+                    .intersperse_with(|| "::")
```

Comment:
Can you do this? If not, can you post the error?

```suggestion
                    .intersperse("::")
```

Reply:
Apparently it does work. I must've misjudged the trait impls somewhere.